api/src/main/java/com/debugmafia/clueless/state/Game.java

```diff
@@ -0,0 +1,144 @@
+            //TODO: Remove SUGGEST and MOVE as available actions for the current players turn
+            //TODO: Set the suggestion in the turn object as the suggestion that was just made
+            }
+
+        }
+        else{
+            //TODO: Remove SUGGEST and MOVE as available actions for the current players turn
+            //TODO: Set the turn state to IN_PROGRESS
+        }
+        return this;
+    }
+
+    public Game makeAccusation(Accusation a) {
+        //Restrictions: The player making the accusation must be the player whose current turn it is
+        Card accusedWeapon = this.board.getAssociatedCard(a.getWeapon());
+        Card accusedLocation = this.board.getAssociatedCard(a.getRoom());
+        Card accusedPiece = this.board.getAssociatedCard(a.getPiece());
+
+        Card arr[] = {accusedWeapon, accusedLocation, accusedPiece};
+
+        Set<Card> accusedCards = new HashSet<>(Arrays.asList(arr));
+
+        if(winningCards.containsAll(accusedCards)) {
+            //TODO: Set the winner
+            //TODO: Set the game state to GameState.COMPLETE
+            //TODO: Set the accusation object within the current players turn to this accusation.
+        }
+        else {
+            //TODO: Remove this player from the list of active players
+            //TODO: Set the accusation object within the current players turn to this accusation.
+            //TODO: Set the available actions within the current players turn to END_TURN.
+            //TODO: Set the turn state within the current players turn to WAITING_FOR_END_TURN
+        }
+
+        return this;
+    }
+
+    public Game makeRebuttal(Rebuttal r) {
+        //Restrictions: The player making the rebuttal must be the person who was requested a rebuttal from. The card in the rebuttal must be one of the cards 
+        //that was made in the original suggestion.
+        //TODO: Set the rebuttal object to the current rebuttal on the current players turn
+        //TODO: Set the turn state to WAITING_FOR_END_TURN on the current players turn
+        //TODO: Set the available actions to only END_TURN on the current players turn
+        return this;
+    }
+
+    public Game endTurn(Player p) {
+        //What is the constructor for Turn object?
+        Turn newTurnInstance = new Turn();
+        //TODO: Find the next active player to take their turn (next in the array)
+        Player nextActivePlayer = this.getNextActivePlayer(p);
+        //TODO: Set that player as the player in the current turn instance
+  
+        BoardLocation location = this.board.getPieceLocation(p.getPiece());
+        Set<BoardLocation> openAdjacentLocations = this.board.getOpenAdjacentLocations(location);
+        //TODO: Set the available locations property on the current turn object to the result from above
+        //TODO: Set this new turn instance to the current players turn
+        return this;
+    }
+
+    private Player getNextActivePlayer(Player currentPlayer) {
```

Comment:
I'm not following how the first move of the game would affect this logic in any way, shape or form.

Reply:
I think we should set up playerList so that they are in correct order (i.e. whomever is supposed to go first will always go first) rather than bake that logic into this function. Further, I don't think currentPlayersTurn should ever be null (because there will always be a turn as long as the game is alive). If anything, maybe we put in a function to get the absolute first player (i.e. array index 0), which we can use at start up and when the list wraps.